File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hash_map = {}
        
        for idx, num in enumerate(nums):
            diff = target - num
            if diff in hash_map:
                return [idx, hash_map[diff]]
            hash_map[num] = idx
        return [-1,-1]                

File: 0001-two-sum/0001-two-sum.go
func twoSum(nums []int, target int) []int {
    hashMap := make(map[int]int)
    for i, v := range nums {
        diff := target - v
        val, exists := hashMap[diff]
        if exists {
            return []int{i, val}
        }
        hashMap[v] = i
    }
    return []int{-1,-1}
}